AppSyncRealTimeClient/Support/AppSyncURLHelper.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2018-2021 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public enum AppSyncURLHelper {
+
+    // A standard app sync domain URL would look like
+    // https://abcdefghijklmnopqrstuvwxyz.appsync-api.us-west-2.amazonaws.com/graphql
+    public static let standardDomainPattern =
+    "^https://\\w{26}.appsync-api.\\w{2}(?:(?:-\\w{2,})+)-\\d.amazonaws.com/graphql$"
+
+    // Check whether the provided GraphQL endpoint has standard appsync domain
+    public static func isStandardAppSyncGraphQLDomain(url: URL) -> Bool {
```

Comment:
should this be `isStandardAppSyncGraphQLEndpoint` or `isStandardAppSyncGraphQLURL`? reasoning is because regex appears to be used against the entire URL. I'm thinking the `w{26}.appsync-api.[region].amazonaws.com` is the domainhost, `/graphql` is a path, `https` is the scheme